src/AP1roG/AP1roGPSESolver.cpp

```diff
@@ -0,0 +1,317 @@
+#include "AP1roG/AP1roGPSESolver.hpp"
+#include <numopt.hpp>
+
+
+namespace GQCG {
+
+
+/*
+ * CONSTRUCTORS
+ */
+/**
+ *  Constructor based on a given @param molecule and Hamiltonian parameters @param ham_par
+ *
+ *  The initial guess for the geminal coefficients is zero
+ */
+AP1roGPSESolver::AP1roGPSESolver(const GQCG::Molecule& molecule, const GQCG::HamiltonianParameters& ham_par) :
```

Comment:
Noticed that the molecule is only passed for the amount of electrons. One could change this in the constructor to simply pass the amount rather than the molecule (creates compatibility with simple files not containing xyz files or basis sets, etc (e.g. FCIDUMP for CI))